std/typecons.d

```diff
@@ -5189,6 +5189,8 @@ mixin template Proxy(alias a)
                 return a.opCmp(b);
             else static if (is(typeof(b.opCmp(a))))
                 return -b.opCmp(a);
+            else static if (isFloatingPoint!ValueType || isFloatingPoint!B)
+                return a < b ? -1 : a > b ? +1 : a == b ? 0 : float.nan;
```

Comment:
See e.g. http://goo.gl/3kxR12 (using C in order to test with more up-to-date optimisers). Note how gcc and clang produce quite different results in places. Also, note how bad some of them are, seeing as they should be a single instruction plus some register bookkeeping